### PR TITLE
Add kcp/1.5 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **kcp/1.5**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/kcp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Windows" and self.options.shared:
+            raise ConanInvalidConfiguration("kcp does not support shared builds on Windows")
```
Comment (SpaceIm) [inline recipes/kcp/all/conanfile.py]: Any reason to not enable WINDOWS_EXPORT_ALL_SYMBOLS in cmake wrapper?

Comment (madebr) [reply recipes/kcp/all/conanfile.py]: That should work fine. 
I opened https://github.com/skywind3000/kcp/pull/254 but it does not like they review and/or accept pr's in a timely manner.

File: recipes/kcp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
```
Comment (uilianries) [inline recipes/kcp/all/conanfile.py]:
```suggestion
            del self.options.fPIC
        del self.settings.compiler.libcxx
        del self.settings.compiler.cppstd
```

Comment (madebr) [reply recipes/kcp/all/conanfile.py]: Thanks! I forgot!

**Pull request merged**